src/test/java/cart/web/cart/CartApiControllerTest.java

```diff
@@ -0,0 +1,147 @@
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.context.jdbc.Sql;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+@Sql(scripts = "classpath:schema.sql")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CartApiControllerTest {
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private CartDao cartDao;
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    @BeforeEach
+    void setUp() {
+        RestAssured.port = port;
+
+        // reset tables
+        jdbcTemplate.update("DELETE FROM cart");
+        jdbcTemplate.update("DELETE FROM product");
+        jdbcTemplate.update("DELETE FROM users");
+
+        // user insert
+        jdbcTemplate.update("INSERT INTO users(email, password) VALUES('a@a.com', 'passwordA')"); // id = 1
+        jdbcTemplate.update("INSERT INTO users(email, password) VALUES('b@b.com', 'passwordB')"); // id = 2
+
+        // product insert
+        jdbcTemplate.update("INSERT INTO product(name, image_url, price) VALUES('상품A', 'image.com/imageA', 10000)"); // id = 1
+        jdbcTemplate.update("INSERT INTO product(name, image_url, price) VALUES('상품B', 'image.com/imageB', 20000)"); // id = 2
+        jdbcTemplate.update("INSERT INTO product(name, image_url, price) VALUES('상품C', 'image.com/imageC', 30000)"); // id = 3
+
+        // cart insert
+        jdbcTemplate.update("INSERT INTO cart(user_id, product_id) VALUES(1, 1)");
+        jdbcTemplate.update("INSERT INTO cart(user_id, product_id) VALUES(1, 2)");
+    }
+
+    @Test
+    void 정상_모든_장바구니_아이템을_조회한다() {
+        RestAssured.given()
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .header("Authorization", authorizationHeader("a@a.com", "passwordA"))
+                .when().get("/api/v1/carts")
+                .then()
+                .statusCode(HttpStatus.OK.value())
+                .body("size()", is(2))
+                .body("[0].name", is("상품A"))
+                .body("[1].name", is("상품B"));
+    }
+
+    @Test
+    void 비정상_모든_장바구니_아이템을_조회한다_유저_정보가_없는_경우() {
+        RestAssured.given()
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .when().get("/api/v1/carts")
+                .then()
+                .statusCode(HttpStatus.BAD_REQUEST.value());
+    }
+
+    @Test
+    void 정상_장바구니_아이템을_추가한다() {
+        // given
+        CartAddRequestDto requestDto = new CartAddRequestDto(3L);
+
+        // when
+        Long addedCartId = RestAssured.given()
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .header("Authorization", authorizationHeader("a@a.com", "passwordA"))
+                .body(requestDto)
+                .when().post("/api/v1/carts")
+
+        // then
```

Comment:
이 질문은 너무 사소한거긴 한데요..!!ㅎㅎ

말씀해주신대로 given, when, then 으로 구분하여 테스트 코드를 작성하는 컨벤션을 지켜보았는데요, 위와 같이 RestAssured를 사용할 땐 메서드 체인 자체에 위 키워드들이 있어서 구분하기가 애매했던 것 같습니다.

RestAssered만 사용하는 테스트라면 차라리 그냥 그 자체로 이미 가독성이 괜찮아보여서 그냥 구분하지 않고 둘텐데 위 테스트처럼 RestAssured만 사용하는게 아니라 추가로 다른 값들을 검증하는 경우 구분하기가 애매하더라구요. 혹시 이런 경우에는 가독성을 높이기 위한 방법을 추천해주실 만한게 있을까요?🤔

Reply:
then() 다음에 extract() 메서드를 사용하면 응답을 추출해낼 수 있어요 :)
그런 다음 아래와 같이 사용해볼 수 있습니다 😄 

```suggestion
        // when
        ExtractableResponse<Response> response = RestAssured.given()
                .contentType(MediaType.APPLICATION_JSON_VALUE)
                .when().delete("/api/v1/products/" + savedProduct.getId())
                .then()
                .extract();
                
       // then         
        assertThat(response.statusCode).isEqualTo(HttpStatus.OK.value());
```